tests/cases/fourslash/tsxCompletion15.ts

```diff
@@ -0,0 +1,43 @@
+/// <reference path='fourslash.ts' />
+//@module: commonjs
+//@jsx: preserve
+
+//// declare module JSX {
+////     interface Element { }
+////     interface IntrinsicElements {
+////     }
+////     interface ElementAttributesProperty { props; }
+//// }
+
+//@Filename: exporter.tsx
+//// export module M {
+////    export declare function SFCComp(props: { Three: number; Four: string }): JSX.Element;
+//// }
+
+//@Filename: file.tsx
+//// import * as Exp from './exporter';
+//// var x1  = <Exp.M.SFCComp></[|/*1*/|]>;
+//// var x2  = <Exp.M.SFCComp></[|Exp./*2*/|]>;
+//// var x3  = <Exp.M.SFCComp></[|Exp.M./*3*/|]>;
+//// var x4  = <Exp.M.SFCComp></[|Exp.M.SFCComp/*4*/|]
+//// var x5  = <Exp.M.SFCComp></[|Exp.M.SFCComp/*5*/|]>;
+//// var x6  = <Exp.M.SFCComp></      [|Exp./*6*/|]>;
+//// var x7  = <Exp.M.SFCComp></[|/*7*/Exp.M.SFCComp|]>;
+//// var x8  = <Exp.M.SFCComp></[|Exp/*8*/|]>;
+//// var x9  = <Exp.M.SFCComp></[|Exp.M./*9*/|]>;
+//// var x10 = <Exp.M.SFCComp></      [|/*10*/Exp.M.Foo.Bar.Baz.Wut|]>;
```

Comment:
```suggestion
//// var x10 = <Exp.M.SFCComp></      [|Exp./*10*/M.Foo.Bar.Baz.Wut|]>;
```

This is the failing test case due to fixed expression depth. Where it is now, `location` is actually the `/` token.

Reply:
I see so if completion was asked at mid expression it will be a the full closing tag with a replacement span of both sides. I’ll update the PR asap

Reply:
Yeah, `location` could be anywhere in the tag name expression so you need to walk up until you either find the whole `tagName` or realize you’re not in one.

Reply:
@andrewbranch is there a utility to check if we are in jsx/tsx file? we might wanna bail if we don have any jsx at all.

I updated the PR, let me know if you don't like my approach.

Reply:
> we might wanna bail if we don't have any jsx elements at all

Good call. `getLanguageVariant(getScriptKind(fileName, host)) === LanguageVariant.JSX`.

> I updated the PR, let me know if you don't like my approach.

I don’t like the arbitrary counter—it’s more work than necessary to get a true negative in most places, and will return a false negative on super super deep closing tags, which should never happen in practice, but still feels kludgy. The only thing you’ll ever hit where you need to keep going up is a PropertyAccessExpression. Just continue as long as you have an Identifier or a PropertyAccessExpression.